provider/source/src/units/data.rs

```diff
@@ -124,20 +106,25 @@ impl crate::IterableDataProviderCached<UnitsDisplayNameV1Marker> for SourceDataP
             length: &str,
             length_patterns: &BTreeMap<String, Patterns>,
         ) -> Result<(), DataError> {
-            let quantities = length_patterns
-                .keys()
-                // TODO: remove this filter once we are supporting all the units categories.
-                // NOTE:
-                //  if this filter is removed, we have to add a filter to remove all the prefixes.
-                .filter_map(|key| {
-                    if let Some(rest) = key.strip_prefix("length-") {
-                        Some(rest)
-                    } else if let Some(rest) = key.strip_prefix("duration-") {
-                        Some(rest)
-                    } else {
-                        None
-                    }
-                });
+            let quantities = length_patterns.keys().filter_map(|key| {
+                // In order to reduce the number of units in the test data,
+                // we only test the length-* and duration-* units.
+                #[cfg(test)]
+                if let Some(rest) = key.strip_prefix("length-") {
+                    Some(rest)
+                } else if let Some(rest) = key.strip_prefix("duration-") {
+                    Some(rest)
+                } else {
+                    None
+                }
+                #[cfg(not(test))]
+                // NOTE: any units should have the category as a prefix which is separated by `-`.
+                //       Therefore, if the `rest` is empty, it means the key is not for a unit.
+                //       In this case, we should return None.
+                //       Example: `length-meter` is a valid key, but `length` is not.
+                //                `power3` is not a valid unit.
+                key.split_once('-').map(|(_, rest)| rest)
```

Comment:
If you do something like
```suggestion
                key.split_once('-').map(|(_prefix, unit_name)| unit_name)
```
the code will be self-explanatory enough that you don't need the comment.